src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
+        }
+        Node nodeByIndex = getNode(index);
+        return (T) nodeByIndex.value;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        Node previousNode;
+        if (index == 0) { //setAsHead
+            previousNode = head;
+
+            newNode.next = head.next;
+            head.next.prev = newNode;
+
+            head = newNode;
+        } else if (index == size - 1) { //addAsTail
+            previousNode = tail;
+
+            newNode.prev = tail.prev;
+            tail.prev.next = newNode;
+
+            tail = newNode;
+        } else {
+            previousNode = getNode(index);
+            Node prevNode = previousNode.prev;
+            Node nextNode = previousNode.next;
+            newNode.prev = prevNode;
+            newNode.next = nextNode;
+            prevNode.next = newNode;
+            nextNode.prev = newNode;
+        }
+        return (T) previousNode.value;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node removedNode;
+        if (index == 0) {
+            removedNode = head;
+            head = head.next;
+            if (head == null) {
+                tail = null;
+            } else {
+                head.prev = null;
+            }
+        } else if (index == size - 1) {
+            removedNode = tail;
+        } else {
+            removedNode = getNode(index);
+        }
+
+        Node prevNode = removedNode.prev;
+        Node nextNode = removedNode.next;
+        if (prevNode != null) {
+            prevNode.next = nextNode;
+        }
+        if (nextNode != null) {
+            nextNode.prev = prevNode;
+        }
+        size--;
+        return (T) removedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node currentNode = head;
+        while (currentNode != null) {
+            if (currentNode.value == object
+                    || currentNode.value != null && currentNode.value.equals(object)) {
+                if (size == 1) {
+                    size--;
+                    return true;
+                }
+                Node prevNode = currentNode.prev;
+                Node nextNode = currentNode.next;
+                if (prevNode == null) {
+                    nextNode.prev = null;
+                    head = nextNode;
+                } else {
+                    prevNode.next = nextNode;
+                }
+                if (nextNode == null) {
+                    prevNode.next = null;
+                    tail = prevNode;
+                } else {
+                    nextNode.prev = prevNode;
+                }
+                size--;
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
```

Comment:
Create a separate method unlink(Node node) for this
It was it in common mistakes 
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html